feat(linter): implement eslint/capitalized-comments

tt-a1i opened the pull request:
Implements ESLint's `capitalized-comments` rule which enforces or disallows capitalization of the first letter of a comment.

## Features

- Support for `"always"` (default) and `"never"` options
- Configuration options:
  - `ignorePattern`: Regex to ignore specific comments
  - `ignoreInlineComments`: Ignore comments on same line as code
  - `ignoreConsecutiveComments`: Ignore comments following another comment
  - Separate configuration for line vs block comments
- Directive comments (eslint-disable, etc.) are automatically ignored
- URLs in comments are automatically ignored
- Auto-fix support to change capitalization

## Test Plan

- [x] Unit tests for pass/fail cases
- [x] Fix tests for auto-fix functionality
- [x] Snapshot test
- [x] `cargo clippy -p oxc_linter` passes
- [x] `cargo test -p oxc_linter capitalized_comments` passes

Part of  #479

tt-a1i committed:
feat(linter): implement eslint/capitalized-comments

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

camc314 committed:
u

camc314 reviewed (changes requested):
I re-ported the test cases from ESLint and some are failing. 

These cases need to be fixed

tt-a1i committed:
fix(linter): fix directive detection in capitalized-comments

Use original content instead of normalized content when checking for
directive comments. This ensures that '//* eslint-disable' is NOT
treated as a directive (because '*' precedes it), while '// eslint-disable'
is correctly recognized as a directive.

This fixes the failing test cases that camc314 ported from ESLint.

camc314 committed:
u

camc314 committed:
u

camc314 reviewed (approved):
THank you

tt-a1i commented:
Good catch — agreed. `eslint/capitalized-comments` should ignore OxLint directives like `oxlint-disable-line` / `oxlint-disable-next-line` so we don’t warn or auto-fix them accidentally.

#16896 is already merged, so I opened a small follow-up PR to add this: https://github.com/oxc-project/oxc/pull/16989

camc314 commented:
> Just checking in here since this was merged recently – was it intentional to make this rule enabled by default?

Can you share your `.oxlintrc.json` ? This rule shouldn't be enabled by default

camc314 commented:
Ah I see, you've got:

```
    "style": "error",
```
Which enables the style category. Since this rule is within the style category, oxlint will start reporting this error.

The style category is off by default.

If you don't want to use this rule, you'd add `"capitalized-comments": "error"` to your rules object

htunnicliff commented:
That makes total sense 🤦‍♂️ – thank you for taking a look! 🙇
